src/components/PostListVolunteer/index.jsx

```diff
@@ -0,0 +1,149 @@
+/* eslint-disable array-callback-return */
+/* eslint-disable consistent-return */
+/* eslint-disable react/jsx-one-expression-per-line */
+/* eslint-disable jsx-a11y/label-has-associated-control */
+/* eslint-disable import/extensions */
+/* eslint-disable prefer-const */
+/* eslint-disable prefer-destructuring */
+
+import React, { lazy, Suspense, useEffect, useState } from "react";
+import { Row, Container } from "react-bootstrap";
+import { useTranslation } from "react-i18next";
+import "./index.css";
+import db from "../../firebaseConfig.js";
+// eslint-disable-next-line no-unused-vars
+import i18n from "../../i18n";
+
+const PostVolunteer = lazy(() => import("../PostVolunteer/index.jsx"));
+
+export default function PostListVolunteer() {
+  const [posts, setPosts] = useState([]);
+  const [secPosts, setSecPosts] = useState([]);
+  const { t } = useTranslation();
+
+  const fetchDataDonator = async () => {
+    let res = await db.collection("donatorPost").get();
+    const data = res.docs.map((post) => {
+      const tmp = post.data();
+      const id = post.id;
+      return { id, ...tmp };
+    });
+
+    setPosts(data);
+    setSecPosts(data);
+  };
+
+  useEffect(() => {
+    fetchDataDonator();
+  }, []);
+
+  const handleSort = async (e) => {
+    if (e.target.value === "title") {
+      let res = await db
+        .collection("donatorPost")
+        .orderBy("title", "asc")
+        .get();
+      const data = res.docs.map((post) => {
+        const tmp = post.data();
+        const id = post.id;
+        return { id, ...tmp };
+      });
+      setPosts(data);
+    } else if (e.target.value === "date") {
+      let res = await db
+        .collection("donatorPost")
+        .orderBy("date", "desc")
+        .get();
+      const data = res.docs.map((post) => {
+        const tmp = post.data();
+        const id = post.id;
+        return { id, ...tmp };
+      });
+      setPosts(data);
+    }
+  };
+
+  const handleType = (e) => {
+    if (e.target.value === "all") {
+      setPosts(secPosts);
+    } else {
+      let fltr = secPosts.filter((el) => el.type === e.target.value);
+      setPosts(fltr);
+    }
+  };
+
+  const handleCity = async (e) => {
+    if (e.target.value === "all") {
+      setPosts(secPosts);
+    } else {
+      let fltr = secPosts.filter((el) => el.address[0] === e.target.value);
+      setPosts(fltr);
+    }
+  };
+
+  const handleSearch = async (e) => {
+    const search = e.target.value;
```

Comment:
Can names be more meaningful here, i.e. searchKeyWord instead of search, and searchResults instead of results